src/Authentication.Abstractions.Test/AzureEnvironmentTests.cs

```diff
@@ -52,6 +52,17 @@ public void TestArmCloudMetadata20190501Init()
             }
         }
 
+        [Fact]
+        public void TestArmResponseNoAzureCloud()
+        {
+            Environment.SetEnvironmentVariable(ArmMetadataEnvVariable, @"TestData\ArmResponseNoAzureCloud.json");
+            var armEnvironments = AzureEnvironment.InitializeBuiltInEnvironments(null, httpOperations: TestOperationsFactory.Create().GetHttpOperations());
+
+            // Check AzureCloud is added to public environment list even discovery endpoint doesn't return AzureCloud.
+            Assert.True(AzureEnvironment.PublicEnvironments.ContainsKey(EnvironmentName.AzureCloud));
```

Comment:
I would assert this for all of the built-in environments

Reply:
Sure. I will add them.